client/src/app/site/agenda/agenda-list/agenda-list.component.ts

```diff
@@ -15,26 +17,44 @@ import { Topic } from '../../../shared/models/topics/topic';
     templateUrl: './agenda-list.component.html',
     styleUrls: ['./agenda-list.component.css']
 })
-export class AgendaListComponent extends BaseComponent implements OnInit {
+export class AgendaListComponent extends ListViewBaseComponent<ViewItem> implements OnInit {
     /**
      * The usual constructor for components
      * @param titleService
      * @param translate
      */
-    public constructor(titleService: Title, protected translate: TranslateService) {
+    public constructor(
+        titleService: Title,
+        translate: TranslateService,
+        private router: Router,
+        private repo: AgendaRepositoryService
+    ) {
         super(titleService, translate);
     }
 
     /**
      * Init function.
-     * Sets the title
+     * Sets the title, initializes the table and calls the repository.
      */
     public ngOnInit(): void {
         super.setTitle('Agenda');
-        // tslint:disable-next-line
-        const i: Item = new Item(); // Needed, that the Item.ts is loaded. Can be removed, if something else creates/uses items.
-        // tslint:disable-next-line
-        const t: Topic = new Topic(); // Needed, that the Topic.ts is loaded. Can be removed, if something else creates/uses topics.
+        this.initTable();
+        this.repo.getViewModelListObservable().subscribe(newAgendaItem => {
+            this.dataSource.data = newAgendaItem;
+        });
+    }
+
+    /**
+     * Handler for click events on agenda item rows
+     * Links to the content object if any
+     */
+    public selectAgendaItem(item: ViewItem): void {
+        if (item.contentObject instanceof Motion) {
```

Comment:
The contentobject is a AgendaBaseModel. So you can get the detail state with .getDetailStateURL()
See agenda-information.ts for the exact name;)
You can also change the signature of this method, if you need something else as a string for the .navigate()
You need to add the url in the motion.ts

Reply:
string will always be okay.
I check for `contentObject` and call `getDetailStateURL` to navigate. `motion.ts` is adjusted.